internal/log/log.go

```diff
@@ -172,3 +175,22 @@ func printMsg(lvl, format string, a ...interface{}) {
 type defaultLogger struct{ l *log.Logger }
 
 func (p *defaultLogger) Log(msg string) { p.l.Print(msg) }
+
+// DiscardLogger discards every call to Log().
+type DiscardLogger struct{}
+
+// Log implements ddtrace.Logger.
+func (d DiscardLogger) Log(msg string) {}
+
+// RecordLogger appends every call to Log() to Logs.
+type RecordLogger struct {
+	m    sync.Mutex
+	Logs []string
```

Comment:
We should have some sort of accessor/getter for this, under the mutex. Otherwise we'll get race conditions in tests if someone tries to log and assert output at the same time...